libraries/support/azure/service.rb

```diff
@@ -84,14 +84,16 @@ def set_reader(name, value, override)
       self
     end
 
-    def get(url:, api_version:, error_handler: nil, unwrap: nil, use_cache: true, params: {}) # rubocop:disable Metrics/ParameterLists
+    def get(url:, api_version:, error_handler: nil, unwrap: nil, use_cache: true, params: {}, headers: {}) # rubocop:disable Metrics/ParameterLists
```

Comment:
Allow extending Clients to pass headers into their requests. Queue API defines version this way.